drivers/serial/uart_nrfx_uart.c

```diff
@@ -239,13 +267,17 @@ static void uart_nrfx_poll_out(struct device *dev,
 	nrf_uart_txd_set(uart0_addr, (u8_t)c);
 
 	/* Wait until the transmitter is ready, i.e. the character is sent. */
-	while (!event_txdrdy_check()) {
-	}
+	int res;
 
-	/* Deactivate the transmitter so that it does not needlessly consume
-	 * power.
+	NRFX_WAIT_FOR(event_txdrdy_check(), 1000, 1, res);
```

Comment:
I don't entirely get the idea behind this change. When can it happen that the transmission of the character does not finish?

Reply:
There is very small risk that if asynchronous TX started in line 937 during poll out in interrupt and ended in lines 964-965 then asynchronous call will clear ENDTX event and result in hanging that poll_out in interrupt.